pylint/checkers/variables.py

```diff
@@ -1316,6 +1316,13 @@ def _check_consumer(
             if utils.is_func_decorator(current_consumer.node) or not (
                 current_consumer.scope_type == "comprehension"
                 and self._has_homonym_in_upper_function_scope(node, consumer_level)
+                # But don't catch homonyms against the filter of a comprehension,
+                # "if x" in [x for x in expr() if x]
```

Comment:
```suggestion
                # if x in [x for x in expr() if x]
```

Reply:
I like the quotes here as it shows that the `if x` is not part of the code but refers to the `if x` in the code example after the first `in`.

Reply:
~~I don't understand at all 😄 ~~ nevermind it took me 30s. 

```suggestion
                # (like "if x" in "[x for x in expr() if x]")
```

Reply:
This suggestion I can get behind 😄